core/src/main/scala/stryker4s/mutants/findmutants/MutantMatcher.scala

```diff
@@ -1,22 +1,23 @@
 package stryker4s.mutants.findmutants
 
+import stryker4s.config.Config
 import stryker4s.extensions.mutationtypes._
 import stryker4s.model.Mutant
 
 import scala.meta.{Term, Tree}
 
-class MutantMatcher {
+class MutantMatcher()(implicit config: Config) {
 
   private[this] val stream = Iterator.from(0)
 
-  def allMatchers(): PartialFunction[Tree, Seq[Mutant]] =
+  def allMatchers(): PartialFunction[Tree, Seq[Option[Mutant]]] =
```

Comment:
Instead of making a `Seq[Option[Mutant]]`, why not still return a `Seq[Mutant]`, but just filter out the mutants that are excluded. So sometimes giving an empty list